src/SpecFuzzPass.cpp

```diff
@@ -633,24 +633,150 @@ auto X86SpecFuzzPass::visitWrite(MachineInstr &MI, MachineBasicBlock &Parent) ->
         .add(MI.getOperand(MemRefBegin + X86::AddrSegmentReg));
 
     restoreRegister(Parent, MI, Loc, X86::RSP, "checkpoint_sp");
-
+	
+	MachineMemOperand *MMO = *MI.memoperands_begin();
+    uint64_t width = MMO->getSize();
+	
+	LLVM_DEBUG(dbgs() << "Store's width: " << width << "\n");
+	
+	BuildMI(Parent, MI, Loc, TII->get(X86::PUSH64i8))
+		.addImm((width > 8)? 8 : width);
+	
     // PUSH %TmpReg
     BuildMI(Parent, MI, Loc, TII->get(X86::PUSH64r), TmpReg);
-
-    // PUSH (%TmpReg)
-    BuildMI(Parent, MI, Loc, TII->get(X86::PUSH64rmm), TmpReg)
-        .addImm(1).addReg(0)
-        .addImm(0).addReg(0);
+	
```

Comment:
That's SIMD support, isn't it? Could you add a comment with an explanation of what's happening here? Also, it looks like it could be moved into a separate function.